### PR TITLE
Generate and implement maps render v2

### PR DESCRIPTION
**Pull request by jecmenicanikola**
This PR contains:

- Generated code for Maps Rendering from stable/2024-04-01/render.json
- Updated samples
- Updated tests
- Upgrade version to 2.0.0-beta.1

### CODE REVIEW

File: sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderAsyncClient.java [hunk 1]
```diff
@@ -175,10 +176,9 @@ Mono<Response<BinaryData>> getMapTileWithResponse(MapTileOptions options, Contex
                 options.getTileIndex(), options.getTimestamp(), options.getMapTileSize(), options.getLanguage(),
                 options.getLocalizedMapView(), context)
             .onErrorMap(throwable -> {
-                if (!(throwable instanceof ErrorResponseException)) {
+                if (!(throwable instanceof ErrorResponseException exception)) {
```
Comment (dubiety) [inline sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderAsyncClient.java:179]: Why do we need `exception` here? Other SDKs don't have `exception` in the statement.

File: sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderClient.java [hunk 2]
```diff
@@ -224,7 +224,7 @@ public Response<MapAttribution> getMapAttributionWithResponse(TilesetId tilesetI
      * <p>
      * Fetches state tiles in vector format typically to be integrated into indoor maps module of map control or SDK.
      * The map control will call this API after user turns on dynamic styling (see [Zoom Levels and Tile
-     * Grid](https://docs.microsoft.com/azure/location-based-services/zoom-levels-and-tile-grid)).
+     * Grid](<a href="https://docs.microsoft.com/azure/location-based-services/zoom-levels-and-tile-grid">...</a>)).
```
Comment (dubiety) [inline sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderClient.java:227]: What's this? Why it's `...` for the text?

File: sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderServiceVersion.java [hunk 3]
```diff
@@ -12,7 +12,7 @@ public enum MapsRenderServiceVersion implements ServiceVersion {
     /**
     * Service version {@code 2.1}.
```
Comment (dubiety) [inline sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderServiceVersion.java:13]: Should the `{@code 2.1}` be updated to `2024-01-01`?

File: sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderServiceVersion.java [hunk 4]
```diff
@@ -31,7 +31,7 @@ public enum MapsRenderServiceVersion implements ServiceVersion {
      * @return the latest {@link MapsRenderServiceVersion}
      */
     public static MapsRenderServiceVersion getLatest() {
-        return V2_1;
+        return V2;
```
Comment (dubiety) [inline sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderServiceVersion.java:34]: Why it's changed from `V2_1` to `V2`? Shouldn't it be `V2024_04_01` if we follow the same naming rule?

**Comment by jecmenicanikola** [comment]
@alzimmermsft Could you run pipeline build?

File: sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderServiceVersion.java (hunk 4 above)
Comment (dubiety) [reply sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderServiceVersion.java:34]: This is how blob storage name their version: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceVersion.java

File: sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderAsyncClient.java (hunk 1 above)
Comment (jecmenicanikola) [reply sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderAsyncClient.java:179]: @dubiety it was suggested from Java

File: sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderServiceVersion.java (hunk 4 above)
Comment (jecmenicanikola) [reply sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderServiceVersion.java:34]: `        internal const ServiceVersion LatestVersion = ServiceVersion.V2_0;

        /// <summary> The version of the service to use. </summary>
        public enum ServiceVersion
        {
            /// <summary> Service version "1.0". </summary>
            V1_0 = 1,
            /// <summary> Service version "2.0". </summary>
            V2_0 = 2,
        }`
        
       here what we did for C#, I followed that

Comment (jecmenicanikola) [reply sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderServiceVersion.java:34]: @dubiety

File: sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderClient.java (hunk 2 above)
Comment (jecmenicanikola) [reply sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderClient.java:227]: It was suggested form javadoc warning, now I added text for this link. @dubiety

File: sdk/maps/azure-maps-render/CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 1.0.0-beta.3 (Unreleased)
+## 2.0.0-beta.1 (Unreleased)
```
Comment (dubiety) [inline sdk/maps/azure-maps-render/CHANGELOG.md:3]: Can we add what's changed in this version?

File: sdk/maps/azure-maps-render/README.md [hunk 6]
```diff
@@ -73,7 +73,7 @@ See [Authentication][authenticate] for more options.
 See [API design][design] for general introduction on design and key concepts on Azure Management Libraries.
 
 ## Examples
-Get Map Tile
+###Get Map Tile
```
Comment (dubiety) [inline sdk/maps/azure-maps-render/README.md:76]:
```suggestion
### Get Map Tile
```

File: sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderServiceVersion.java (hunk 4 above)
Comment (dubiety) [reply sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderServiceVersion.java:34]: I think we should consult Java SDK archboard.

File: sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderClient.java (hunk 2 above)
Comment (dubiety) [reply sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderClient.java:227]: Sorry, what text? I'm a little bit confused

Comment (jecmenicanikola) [reply sdk/maps/azure-maps-render/src/main/java/com/azure/maps/render/MapsRenderClient.java:227]: @dubiety text inside <a> tag

**Review by dubiety — APPROVED** [review]

**Pull request merged**